pynumdiff/utils/simulate.py

```diff
@@ -11,129 +11,85 @@
 
 
 # pylint: disable-msg=too-many-locals, too-many-arguments, no-member
-def __add_noise__(x, noise_type, noise_parameters, random_seed):
-    """
-    Adding synthetic noise to the time series data
-
-    :param x: an array of time series data
-    :type x: np.array
-    :param noise_type: a string representing type of noise, should be compatible with np.random functions
-                       (eg. 'normal', 'uniform', 'poisson')
-    :type noise_type: string
-    :param noise_parameters: a list of parameters of the noise used in np.random
-    :type noise_parameters: list
-    :param random_seed: an integer seed used to initialize the random number generator
-    :type random_seed: int
-
-    :return: noisy time series data
-    :rtype: np.array
-    """
-    _np.random.seed(random_seed)
-    timeseries_length = _np.max(x.shape)
-    noise = _np.random.__getattribute__(noise_type)(noise_parameters[0], noise_parameters[1],
```

Comment:
Passing `noise_parameters[0], noise_parameters[1]` only works if there are two parameters, like for `normal`. For `poisson` this wouldn't work, so I've changed this to be variable arguments with a *.